libsolidity/ast/Types.cpp

```diff
@@ -1264,6 +1264,42 @@ FixedPointType const* RationalNumberType::fixedPointType() const
 	);
 }
 
+pair<RationalNumberType const*, RationalNumberType const*> RationalNumberType::fractionalDecomposition() const
+{
+	rational const maxUint = rational((bigint(1) << 256) - 1);
+	rational const minInt = -rational(bigint(1) << 255);
+
+	bool negative = (m_value < 0);
+	rational const maxMantissa = (negative ? -minInt : maxUint);
+
+	rational negatedExponent = 0;
+	rational unsignedMantissa = abs(m_value);
+
+	if (unsignedMantissa > maxMantissa)
+		return {nullptr, TypeProvider::rationalNumber(0)};
+
+	while (unsignedMantissa.denominator() != 1)
+	{
+		unsignedMantissa *= 10;
+		++negatedExponent;
+
+		// NOTE: The exponent calculated here can be arbitrarily large because we do not limit precision
+		// of rational literals. While the language does not allow very large exponents in scientific
+		// notation, the number of significant digits in the literal itself is unlimited.
```

Comment:
Please just remove this comment. There is zero question about why we impose limits here - those limits have no connection whatsoever on the completely unrelated limits you're referring to. This comment is still more confusing than helpful - without it, the code is perfectly clear and fine.

Reply:
I strongly disagree but I don't want to continue this discussion so I removed it.